massa-grpc/src/api.rs

```diff
@@ -339,6 +339,30 @@ pub(crate) fn get_largest_stakers(
     })
 }
 
+// Get node version
+pub(crate) fn get_mip_status(
+    grpc: &MassaGrpc,
+    request: tonic::Request<grpc::GetMipStatusRequest>,
+) -> Result<grpc::GetMipStatusResponse, GrpcError> {
+    let mip_store_status_ = grpc.mip_store.get_mip_status();
+    let mip_store_status: Result<Vec<grpc::MipStatusEntry>, GrpcError> = mip_store_status_
+        .iter()
+        .map(|(mip_info, state_id)| {
+            Ok(grpc::MipStatusEntry {
+                mip_info: Some(grpc::MipInfo::from(mip_info)),
+                state: i32::try_from(u32::from(state_id.clone())).map_err(|_e| {
+                    GrpcError::InvalidArgument("Cannot convert state_id".to_string())
```

Comment:
I'm a fan of giving the contexte in error message: always give a complete or partial part of this invalid argument.

Reply:
Done